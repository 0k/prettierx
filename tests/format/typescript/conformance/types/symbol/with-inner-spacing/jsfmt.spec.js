// [prettierx] test script notice:
// This test script runs for test files in parent directory,
// **not** on any files in *this* directory.

const dirPath = `${__dirname}/..`;

run_spec(dirPath, ["typescript"], {
  // [prettierx] test with spacing options (...)
  typeAngleBracketSpacing: true,
  typeBracketSpacing: true,
});
