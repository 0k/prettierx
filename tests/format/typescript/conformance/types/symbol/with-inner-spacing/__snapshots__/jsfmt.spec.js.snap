// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`minustoken.ts 1`] = `
====================================options=====================================
parsers: ["typescript"]
printWidth: 80
typeAngleBracketSpacing: true
typeBracketSpacing: true
                                                                                | printWidth
=====================================input======================================
type MutableRequired<T> = { 
-readonly [P in keyof T]-?:T[P] 
};  // Remove readonly and ?

type ReadonlyPartial<T> = {
+readonly [P in keyof T]+?:T[P] 
};  // Add readonly and ?

=====================================output=====================================
type MutableRequired< T > = {
  -readonly [ P in keyof T ]-?: T[ P ];
}; // Remove readonly and ?

type ReadonlyPartial< T > = {
  +readonly [ P in keyof T ]+?: T[ P ];
}; // Add readonly and ?

================================================================================
`;

exports[`question.ts 1`] = `
====================================options=====================================
parsers: ["typescript"]
printWidth: 80
typeAngleBracketSpacing: true
typeBracketSpacing: true
                                                                                | printWidth
=====================================input======================================
var x: {
    [A in keyof B]?: any;
};

=====================================output=====================================
var x: {
  [ A in keyof B ]?: any;
};

================================================================================
`;

exports[`readonly.ts 1`] = `
====================================options=====================================
parsers: ["typescript"]
printWidth: 80
typeAngleBracketSpacing: true
typeBracketSpacing: true
                                                                                | printWidth
=====================================input======================================
var x: {
    readonly [A in keyof B]: any;
};

=====================================output=====================================
var x: {
  readonly [ A in keyof B ]: any;
};

================================================================================
`;

exports[`symbol.ts - {"typeAngleBracketSpacing":true,"typeBracketSpacing":true} format 1`] = `
====================================options=====================================
parsers: ["typescript"]
printWidth: 80
typeAngleBracketSpacing: true
typeBracketSpacing: true
                                                                                | printWidth
=====================================input======================================
var x: symbol
var x: symbol | string

=====================================output=====================================
var x: symbol;
var x: symbol | string;

================================================================================
`;
